File: Backend/PlayNow.CleanArchitecture/PlayNow.Domain/Entities/Usuario.cs
namespace PlayNow.Domain.Entities;

public partial class Usuario
{
    public int IdUsuario { get; set; }

    public string Nome { get; set; }

    public string Email { get; set; }

    public string Senha { get; set; }

    public string? Telefone { get; set; }

    public bool IsAdmin { get; set; }

    public bool Deletado { get; set; }
}

File: Frontend/lib/domain/usuario.dart
import 'package:play_now/domain/reserva.dart';

abstract class Usuario {
  int id;
  String nome;
  String email;
  String senha;
  String telefone;
  bool isAdmin;

  Usuario({
    required this.id,
    required this.nome,
    required this.email,
    required this.senha,
    required this.telefone,
    required this.isAdmin,
  });

  void fazerReserva(Reserva reserva) {}

  void cancelarReserva(Reserva reserva) {}
}
